### ISSUE TITLE
Async execution model memory leak

### ISSUE THREAD

**Issue by unleashed** [3scale/apisonator#308]
It looks like when enabling async Apisonator leaks memory. This _might_ be fixed in upstream dependencies of the async reactor, but we are blocked on #303.

@3scale/operations  can you provide more data on this? Grafana dashboard screen captures would be nice to have.

**Comment by slopezz** [comment]
This is an example of the memory usage by backend-worker pods, how memory grows to infinite:
![image](https://user-images.githubusercontent.com/41513123/131833670-ab514583-3985-4657-a69f-f7ad1ab0706b.png)

Our memory limit is 1GB, because of arriving to the memory limit (we have alerts when 90% of a mem/cpu limit is reached), we need to recycle the backend-worker pods every 3-4 days.

Do you need anything else? @unleashed 

AFAIK https://github.com/3scale/apisonator/issues/303 shouldnt be a blocker anymore, because legacy infra is already decommissioned.
